network/multiplexer.go

```diff
@@ -24,49 +24,74 @@ import (
 // Multiplexer is a message handler that sorts incoming messages by Tag and passes
 // them along to the relevant message handler for that type of message.
 type Multiplexer struct {
-	msgHandlers atomic.Value // stores map[Tag]MessageHandler, an immutable map.
+	msgHandlers   atomic.Value // stores map[Tag]MessageHandler, an immutable map.
+	msgProcessors atomic.Value // stores map[Tag]MessageProcessor, an immutable map.
 }
 
 // MakeMultiplexer creates an empty Multiplexer
 func MakeMultiplexer() *Multiplexer {
 	m := &Multiplexer{}
-	m.ClearHandlers([]Tag{}) // allocate the map
+	m.ClearHandlers(nil)   // allocate the map
+	m.ClearProcessors(nil) // allocate the map
 	return m
 }
 
-// getHandlersMap retrieves the handlers map.
-func (m *Multiplexer) getHandlersMap() map[Tag]MessageHandler {
-	handlersVal := m.msgHandlers.Load()
-	if handlers, valid := handlersVal.(map[Tag]MessageHandler); valid {
+// getMap retrieves a typed map from an atomic.Value.
+func getMap[T any](source *atomic.Value) map[Tag]T {
```

Comment:
While many of the functions in multiplexer.go are simple enough, basic unit tests would have been good to include for things like registering/deregistering with filters, etc.

Reply:
it has 60% coverage via `TestMultiplexer` (missing one-liner `Validate`, `Process`, `RegisterProcessors` and `clearMultiplexer`), and 95% via the remaining network tests.